### PR TITLE
Handle asset prefix better

### PR DESCRIPTION
**Pull request by vkalpias**
Added asset.file.originalUrl which equals the old asset.file.url. Instead the asset.file.url is now patched at runtime to contain the asset registry's prefix.

This is to make sure that asset.file.url always points to the full URL that the asset can be found at (even when the assets are proxied from a different domain).

### CODE REVIEW

**Comment by Maksims** [comment]
Changing `file.url` at runtime will lead to problems.
Have we engaged another option by making a method `getFullUrl` that will return full path (including host), instead of modifying `file.url`?

**Comment by vkalpias** [comment]
`file.url` is used in various projects to reference assets. This makes sure that this will continue working when apps are proxied.

**Comment by vkalpias** [comment]
So I changed this to use asset.getFilrUrl instead because it makes the code a lot cleaner. We will need to update projects that use asset.file.url though like tanx

File: src/asset/asset-registry.js [hunk 1]
```diff
@@ -352,8 +344,9 @@ pc.extend(pc, function () {
 
                     self.fire("load", asset);
                     self.fire("load:" + asset.id, asset);
-                    if (asset.file && asset.file.url) {
-                        self.fire("load:url:" + asset.file.url, asset);
+                    var assetFileUrl = asset.getFileUrl();
+                    if (assetFileUrl) {
+                        self.fire("load:url:" + assetFileUrl, asset);
```
Comment (vkalpias) [inline src/asset/asset-registry.js]: Not too sure about which URL to use here, the asset.file.url or the full URL with the prefix.

**Pull request merged**